types/wx-js-sdk-dt/index.d.ts

```diff
@@ -0,0 +1,568 @@
+         */
+        debug?: boolean;
+
+        /**
+         * 公众号的唯一标识
+         */
+        appId: string;
+
+        /**
+         * 签名的时间戳
+         */
+        timestamp: number;
+
+        /**
+         * 签名的随机串
+         */
+        nonceStr: string;
+
+        /**
+         * 签名
+         */
+        signature: string;
+
+        /**
+         * 需要使用的JS接口列表
+         */
+        jsApiList: string[];
+    }
+
+    /**
+     * 通过 config 接口注入权限验证配置
+     * @param setting 开启调试模式,调用的所有 api 的返回值会在客户端 alert 出来，若要查看传入的参数，可以在 pc 端打开，参数信息会通过 log 打出，仅在 pc 端时才会打印。
+     * jsApiList[] 需要使用的 JS 接口列表
+     */
+    function config(setting: wxconfig): void;
+
+    /**
+     * config 信息验证后会执行 ready 方法，所有接口调用都必须在 config 接口获得结果之后，config 是一个客户端的异步操作，所以如果需要在页面加载时就调用相关接口，则须把相关接口放在 ready 函数中调用来确保正确执行。对于用户触发时才调用的接口，则可以直接调用，不需要放在 ready 函数中。
+     * @param x
+     */
+    function ready(x: () => void ): void;
+
+    /**
+     * config 信息验证失败会执行 error 函数，如签名过期导致验证失败，具体错误信息可以打开 config 的 debug 模式查看，也可以在返回的 res 参数中查看，对于 SPA 可以在这里更新签名。
+     */
+    function error(err: (res: any) => void): void;
+
+    /**
+     * 判断当前客户端版本是否支持指定 JS 接口, checkJsApi 接口是客户端6.0.2新引入的一个预留接口，第一期开放的接口均可不使用 checkJsApi 来检测
+     */
+    interface CheckApiConfig {
+        /**
+         * 需要检测的JS接口列表
+         */
+        jsApiList: string[];
+
+        /**
+         * 以键值对的形式返回，可用的 api 值 true，不可用为 false
+         * 如：{"checkResult":{"chooseImage":true},"errMsg":"checkJsApi:ok"}
+         */
+        success(res: { checkResult: { [methodName: string]: boolean }, errMsg: { msg: string } }): void;
+    }
+
+    /**
+     * 判断当前客户端版本是否支持指定 JS 接口
+     * @param setting CheckApiConfig 对象
+     */
+    function checkJsApi(setting: CheckApiConfig): void;
+
+    interface ShareTimelineConfig {
+        /**
+         * 分享标题
+         */
+        title: string;
+
+        /**
+         * 分享链接，该链接域名或路径必须与当前页面对应的公众号 JS 安全域名一致
+         */
+        link: string;
+
+        /**
+         * 分享图标
+         */
+        imgUrl?: string;
+        success?(): void;
+        cancel?(): void;
+    }
+
+    /**
+     * 获取“分享到朋友圈”按钮点击状态及自定义分享内容接口
+     *
+     * @param setting ShareTimelineConfig 对象
+     */
+    function onMenuShareTimeline(setting: ShareTimelineConfig): void;
+
+    /**
+     * 消息分享对象
+     */
+    interface SharedAppMessage extends ShareTimelineConfig {
+        /**
+         * 分享描述
+         */
+        desc: string;
+
+        /**
+         * 分享类型, music、video 或 link，不填默认为 link
+         */
+        type?: string;
+
+        /**
+         * 如果 type 是 music 或 video，则要提供数据链接，默认为空
+         */
+        dataUrl?: string;
+    }
+
+    /**
+     * 获取“分享给朋友”按钮点击状态及自定义分享内容接口
+     * @param setting title 标题，desc 分享描述，link 分享链接，该链接域名或路径必须与当前页面对应的公众号 JS 安全域名一致， imgUrl 分享图标，type 分享类型, music、video 或 link，不填默认为 link
+     * dataUrl 如果 type 是 music 或 video，则要提供数据链接，默认为空, success 用户确认分享后执行的回调函数, cancel 用户取消分享后执行的回调函数
+     */
+    function onMenuShareAppMessage(setting: SharedAppMessage): void;
```

Comment:
There's no need to describe `setting `'s property again, intellisense will automatically pick them up from `SharedAppMessage `. and it seems some docs should be described in `ShareTimelineConfig`: `success` and `cancel`.